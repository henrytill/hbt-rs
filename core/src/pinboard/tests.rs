use super::*;

fn load_fixture(filename: &str) -> String {
    std::fs::read_to_string(format!(
        "{}/src/pinboard/fixtures/{}",
        env!("CARGO_MANIFEST_DIR"),
        filename
    ))
    .unwrap_or_else(|_| panic!("Failed to load fixture: {}", filename))
}

macro_rules! snapshot_xml_test {
    ($name:ident) => {
        #[test]
        fn $name() {
            let fixture_name = stringify!($name).strip_prefix("test_").unwrap_or(stringify!($name));
            let xml = load_fixture(&format!("{}.xml", fixture_name));
            let posts = Post::from_xml(&xml).unwrap();
            insta::assert_yaml_snapshot!(fixture_name, posts);
        }
    };
}

macro_rules! snapshot_json_test {
    ($name:ident) => {
        #[test]
        fn $name() {
            let fixture_name = stringify!($name).strip_prefix("test_").unwrap_or(stringify!($name));
            let json = load_fixture(&format!("{}.json", fixture_name));
            let posts = Post::from_json(&json).unwrap();
            insta::assert_yaml_snapshot!(fixture_name, posts);
        }
    };
}

snapshot_xml_test!(test_empty);
snapshot_xml_test!(test_xml_sample);
snapshot_json_test!(test_json_sample);
