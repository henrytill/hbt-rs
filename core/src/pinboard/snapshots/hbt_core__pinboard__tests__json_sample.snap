---
source: core/src/pinboard/tests.rs
expression: posts
---
- href: "https://github.com/janestreet/magic-trace"
  time: "2022-04-23T00:29:36Z"
  description: "janestreet/magic-trace: magic-trace collects and displays high-resolution traces of what a process is doing"
  extended: tragically intel-only
  tags:
    - performance
    - profiling
    - tools
  hash: 54dab27be2409c987bb17fc06e47a729
  shared: true
  toread: false
- href: "https://www.intel.com/content/www/us/en/developer/tools/oneapi/vtune-profiler.html#gs.x8oazh"
  time: "2022-04-13T13:12:10Z"
  description: Fix Performance Bottlenecks with Intel® VTune™ Profiler
  extended: ~
  tags:
    - performance
    - profiling
    - tools
  hash: 2ab1611711c8bb5ed9273b8f4b612fca
  shared: false
  toread: true
- href: "http://kcachegrind.sourceforge.net/html/Home.html"
  time: "2022-04-13T08:01:32Z"
  description: ~
  extended: ~
  tags:
    - performance
    - profiling
    - tools
  hash: 0850c315d075a67430db267214b18d13
  shared: false
  toread: false
